### PR TITLE
Bluetooth: Audio: Unchecked return value in audio.h

### LINKED ISSUE #74733: [Coverity CID: 392513] Unchecked return value in subsys/bluetooth/audio/shell/audio.h
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/dcf42917c550714d2457947538b9e29d083e872e/subsys/bluetooth/audio/shell/audio.h

Category: Error handling issues
Function: `print_ltv_array`
Component: Bluetooth
CID: [392513](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392513)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/dcf42917c550714d2457947538b9e29d083e872e/subsys/bluetooth/audio/shell/audio.h#L287

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=392513

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by babrsn**
Unchecked return value in functions: print_ltv_array (Line 287)

Fixes #74733

### CODE REVIEW

**Comment by babrsn** [comment]
Since `print_ltv_elem` always returns true, the data parser can be simply cast to void.

File: subsys/bluetooth/audio/shell/audio.h [hunk 1]
```diff
@@ -284,7 +284,7 @@ static void print_ltv_array(const struct shell *sh, size_t indent, const uint8_t
 		.indent = indent,
 	};
 
-	bt_audio_data_parse(ltv_data, ltv_data_len, print_ltv_elem, &ltv_info);
+	(void)bt_audio_data_parse(ltv_data, ltv_data_len, print_ltv_elem, &ltv_info);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/audio.h:287]: `bt_audio_data_parse` may still return an error even if `print_ltv_elem` never does (in the case that we receive invalid LTV encoded data from a peer device. 

I think a better solution would actually be to output that to the shell as an error

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/shell/audio.h (hunk 1 above)
Comment (babrsn) [reply subsys/bluetooth/audio/shell/audio.h]: Done.

File: subsys/bluetooth/audio/shell/audio.h [hunk 2]
```diff
@@ -283,8 +283,12 @@ static void print_ltv_array(const struct shell *sh, size_t indent, const uint8_t
 		.cnt = 0U,
 		.indent = indent,
 	};
+	int err;
 
-	bt_audio_data_parse(ltv_data, ltv_data_len, print_ltv_elem, &ltv_info);
+	err = bt_audio_data_parse(ltv_data, ltv_data_len, print_ltv_elem, &ltv_info);
+	if (err != 0 && err != -ECANCELED) {
+		shell_print(sh, "ould not parse the data: %d", err);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/audio.h:290]:
```suggestion
		shell_error(sh, "%*sInvalid LTV data: %d", indent, "", err);
```

Suggest to print as error. Also take the indent level into account.

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**